### ISSUE TITLE
BOI Mobile (Bank of India official app), absolutely does not work

### ISSUE THREAD

**Issue by rusticadam** [PrivSec-dev/banking-apps-compat-report#329]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### App name

BOI Mobile

### Link to app

https://play.google.com/store/apps/details?id=com.boi.ua.android

### App version

v16.6.5

### Country of the app

India

### Build Number

UP1A.231105.003.2023110700

### Device list

Pixel 7

### Profile app tested in

Owner profile, Secondary profile(s)

### Google Play installed

Installed

### Google Play services Network permission revoked?

- [ ] Revoked
- [X] Not revoked
- [ ] I did not have Google Play services installed

### Native code debugging

- [ ] Enabled
- [X] Disabled

### Exploit protection compatibility mode

- [X] Enabled
- [X] Disabled

### Stock OS compatibility

- [X] Works
- [ ] Does not work
- [ ] Not tested

### NFC payments

- [ ] Works
- [ ] Works but requires another service
- [ ] Does not work
- [X] N/A (Not supported by app)
- [ ] Not tested

### Description of the app's functionality

This app absolutely does not work in Graphene OS. It requires Mobile Data from the SIM card that is registered with bank KYC (not wifi), precise location, Phone and SMS permissions. Even after granting all these permissions the app will have errors due to SUPL and PSDS being set to GrapheneOS proxy, changing it to standard server does not help either. Some times it will have issues with Internet Connectivity Checks servers. Overall I could not make it work with all the possible tweaks I could think of.

### Are there any extra notes you think users should know about?

The app works with Stock OS and with other vendors like Smasung without any issues, but in general the app devs implemented so many unnecessary checks that it is almost unuseful even in stock OS if someone travels abroad.

**Comment by spring-onion** [comment]
Incorporated too.

**Comment by rusticadam** [comment]
Yes, I just tried it.
Update:
Enabling native code debugging helps the app to open. But the issue arises to access the location. To mitigate the issue, I tried to change both SUPL and PSDS location services to Standard server, turned on WiFi scanning and Bluetooth scanning too. It did not help the app.
In summary, native code debugging helps the app to open, but refuses to register an user due to Location access issue.
